keps/sig-storage/4547-in-tree-volume-plugins-remove-dynamic-provisioning-support/README.md

```diff
@@ -0,0 +1,854 @@
+- Which external provisioner will be used to support dynamic provisioning for testing and development purposes. It is up to the user to choose the appropriate external provisioner.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+N/A
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+Currently, the PersistentVolume binder controller in kube-controller-manager is 
+responsible for binding a Persistent Volume Claim (PVC) to a Persistent Volume (PV).
+During the binding process, if no suitable PV is found, the controller will 
+dynamically provision a new PV. 
+
+The dynamic provisioning support is implemented in the in-tree volume plugins. The 
+kube-controller-manager component has command line flags to enable or disable dynamic 
+provisioning for in-tree volume plugins.
+
+- The `--enable-hostpath-provisioner` flag enable HostPath PV provisioning when running 
+  without a cloud provider. This allows testing and development of provisioning features. 
+  HostPath provisioning is not supported in any way, won't work in a multi-node cluster, 
+  and should not be used for anything other than testing or development.
+- The `--enable-dynamic-provisioning` flag enable dynamic provisioning for environments 
+  that support it.
+
+Once the migration plan of the portworxVolume plugin is completed, only the hostPath
+in-tree volume plugin will support dynamic provisioning. The dynamic provisioning support
```

Comment:
Are there tests currently using the in-tree hostpath plugin for dynamic provisioning?

Reply:
No.

For e2e tests, both hostPathDriver and hostPathSymlinkDriver don't implement the storageframework.DynamicPVTestDriver interface in the `test/e2e/storage/drivers/in_tree.go`.

For unit tests, mockVolumePlugin defined in `pkg/controller/volume/persistentvolume/framework_test.go` is used.

For intergation tests, FakeVolumePlugin defined in `pkg/volume/testing/testing.go` is used in `test/integration/volume/persistent_volumes_test.go`